From 53c7f227fbe518c5125887401417da2fbf4c1380 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 1 Aug 2024 11:52:37 +0200
Subject: [PATCH 1125/1695] drivers: stm32_remoteproc: use
 firewall_dt_get_alternate_conf API for stm32mp15

As the API is available for the stm32mp1 enable the memory protection,
reusing code implemented for the stm32mp2x series.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 98b23392a55ca18ec317010767ce721d3f758b98 drivers: stm32_remoteproc: use firewall_dt_get_alternate_conf API for stm32mp15]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I5acf114093d4100bc006aec2cf0c2735baf9d10f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/396465
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/remoteproc/stm32_remoteproc.c | 21 ---------------------
 1 file changed, 21 deletions(-)

diff --git a/core/drivers/remoteproc/stm32_remoteproc.c b/core/drivers/remoteproc/stm32_remoteproc.c
index 026158c48..420e4800a 100644
--- a/core/drivers/remoteproc/stm32_remoteproc.c
+++ b/core/drivers/remoteproc/stm32_remoteproc.c
@@ -128,7 +128,6 @@ void *stm32_rproc_get(uint32_t rproc_id)
 	return rproc;
 }
 
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 /* Re-apply default access right on the memory regions */
 static TEE_Result
 stm32_rproc_release_mems_access(struct stm32_rproc_instance *rproc)
@@ -190,26 +189,6 @@ err:
 
 	return res;
 }
-#else
-/* Re-apply default access right  on the memory regions */
-static TEE_Result
-stm32_rproc_release_mems_access(struct stm32_rproc_instance *rproc __unused)
-{
-	/* To implement for the stm32mp1 */
-
-	return TEE_SUCCESS;
-}
-
-/* Get the exclusive access on the memory regions */
-static TEE_Result
-stm32_rproc_get_mems_access(struct stm32_rproc_instance *rproc __unused)
-{
-	/* To implement for the stm32mp1 */
-	IMSG("Warning: the remoteproc memories are not protected by firewall");
-
-	return TEE_SUCCESS;
-}
-#endif
 
 static TEE_Result stm32mp2_rproc_start(struct stm32_rproc_instance *rproc)
 {
-- 
2.47.2

